opentelemetry-sdk/src/trace/span_processor_with_async_runtime.rs

```diff
@@ -233,27 +238,32 @@ impl<E: SpanExporter, R: RuntimeChannel> BatchSpanProcessorInternal<E, R> {
         match message {
             // Span has finished, add to buffer of pending spans.
             BatchMessage::ExportSpan(span) => {
-                self.spans.push(span);
+                if self.spans.len() == self.config.max_export_batch_size {
+                    // Replace the oldest span with the new span to avoid suspending messages
+                    // processing.
+                    self.spans.pop_front();
+                }
+                self.spans.push_back(span);
 
                 if self.spans.len() == self.config.max_export_batch_size {
                     // If concurrent exports are saturated, wait for one to complete.
                     if !self.export_tasks.is_empty()
                         && self.export_tasks.len() == self.config.max_concurrent_exports
                     {
+                        // TODO: Refactor to avoid stopping message processing to not delay
+                        //       shutdown/resource set because of export saturation.
                         self.export_tasks.next().await;
                     }
 
-                    let export_result = self.export().await;
-                    let task = async move {
-                        if let Err(err) = export_result {
```

Comment:
There's additional minor changes moving the export execution inside `FuturesUnordered`.

Before these changes the export suspend the `BatchSpanProcessor` execution (meaning the `max_concurrent_exports` setting not work).

Reply:
Rollback. See https://github.com/open-telemetry/opentelemetry-rust/pull/2793#pullrequestreview-2679184631